File: dotnet/test/HelloWorldTests/HelloWorldTests.cs
namespace Hello.World;

public class HelloWorldTests
{
    [Fact]
    public void PrintGreetingWithoutNameTest()
    {
        string[] args = [];
        string expectedOutput = "Hello World from Dotnet !";
        string actualOutput = HelloWorld.PrintGreeting(args);
        Assert.Equal(actualOutput, expectedOutput);
    }

    [Fact]
    public void PrintGreetingWithNameTest()
    {
        string[] args = ["Alter", "Sabeh"];
        string expectedOutput = "Hi Alter Sabeh, greetings from Dotnet !";
        string actualOutput = HelloWorld.PrintGreeting(args);
        Assert.Equal(actualOutput, expectedOutput);
    }

    [Fact]
    public void PrintDateTest()
    {
        DateTime now = DateTime.Now;
        string formattedDate = now.ToString("MMMM d, yyyy HH:mm");
        string expectedOutput = $"Today is {formattedDate}";
        string actualOutput = HelloWorld.PrintDate();
        Assert.Equal(actualOutput, expectedOutput);
    }
}

File: gradle/src/test/kotlin/hello/HelloWorldTests.kt
package hello

import org.apache.commons.lang3.time.DateFormatUtils
import kotlin.test.Test
import kotlin.test.assertEquals

class HelloWorldTests {
    @Test
    fun `print greeting without name test`() {
        val args = emptyArray<String>()
        val expectedOutput = "Hello World from Gradle !"
        val actualOutput = printGreeting(args)
        assertEquals(actualOutput, expectedOutput)
    }

    @Test
    fun `print greeting with name test`() {
        val args = arrayOf("Alter", "Sabeh")
        val expectedOutput = "Hi Alter Sabeh, greetings from Gradle !"
        val actualOutput = printGreeting(args)
        assertEquals(actualOutput, expectedOutput)
    }

    @Test
    fun `print current date test`() {
        val now = System.currentTimeMillis()
        val formattedDate = DateFormatUtils.format(now, "MMMM d, yyyy 'at' hh:mm a")
        val expectedOutput = "Today is $formattedDate"
        val actualOutput = printDate()
        assertEquals(actualOutput, expectedOutput)
    }
}
